ivre/db/mongo.py

```diff
@@ -721,6 +721,52 @@ def getscreenshot(port):
         if url == "field":
             return port.get('screendata')
 
+    def set_label(self, flt, group, label, archives=False):
+        """Adds `label` for the session `group` to every host matching `flt`"""
+        colname = self.colname_oldhosts if archives else self.colname_hosts
+        for host in self.find(colname, flt):
+            labels = host.setdefault('labels', [])
+            try:
+                g_label = [lab for lab in labels if lab['group'] == group][0]
```

Comment:
Do you really need to process the whole `labels` list when the first one might be the one you're looking for? If you want to keep a comprehension construction (and you know I won't blame you for that), use:
```Python
g_label = (lab for lab in labels if lab['group'] == group).next()
```
Obviously the exception you want to catch after that will be `StopIteration`.